### PR TITLE
endpoint for retrieving article by id added

### PR DESCRIPTION
**Pull request by philjacks**
- added endpoint for retrieving article by id
- errors 404 & 400 handled
- tested to make sure correct data is in response

### CODE REVIEW

File: __tests__/get.test.js [hunk 1]
```diff
@@ -25,3 +25,40 @@ describe("getTopics", () => {
       });
   });
 });
+
+describe("GET /api/article/:article_id", () => {
```
Comment (bainesface) [inline __tests__/get.test.js:29]: lil typo ... make sure your description pluralises `articles`

File: models/articlesModels.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const db = require("../db/connection");
+
+exports.fetchArticleByIdFromDb = (id) => {
+  const validId = /[0-9]+/.test(id);
+
+  if (!validId) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
```
Comment (bainesface) [inline models/articlesModels.js:8]: I enjoy that you are validating your id but psql will throw an error for us if our datatype does not follow the constraints of the table set up!

File: __tests__/get.test.js (hunk 1 above)
Comment (philjacks) [reply __tests__/get.test.js]: fixed :)

File: models/articlesModels.js (hunk 2 above)
Comment (philjacks) [reply models/articlesModels.js]: I've now changed it to handle the correct error :)

**Pull request merged**